libcontainer/cgroups/fs2/fs2.go

```diff
@@ -206,10 +209,42 @@ func (m *manager) Set(container *configs.Config) error {
 	if err := setFreezer(m.dirPath, container.Cgroups.Freezer); err != nil {
 		return err
 	}
+	if err := m.setUnified(container.Cgroups.Unified); err != nil {
+		return err
+	}
 	m.config = container.Cgroups
 	return nil
 }
 
+func (m *manager) setUnified(res map[string]string) error {
+	for k, v := range res {
+		if strings.Contains(k, "/") {
+			return fmt.Errorf("unified resource %q must be a file name (no slashes)", k)
+		}
+		if err := fscommon.WriteFile(m.dirPath, k, v); err != nil {
+			errC := errors.Cause(err)
+			// cgroup2 returns EPERM (rather than ENOENT) for
```

Comment:
is EPERM because the file is opened with `O_CREAT`?

Reply:
Right. `fscommon.WriteFile` use `ioutil.WriteFile` which calls `os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)`.

Let me fix the comment.

Wonder if we should not use `O_CREAT` as it makes no sense for cgroupfs.

Reply:
fixed